subjects/graphql/README.md

```diff
@@ -0,0 +1,185 @@
+## GraphQL
+
+### Objectives
+
+The objective of this project is to learn graphql query language by creating profiles. It will be provided,\
+by the platform, a graphql endpoint that is connected to the database. So you can query this endpoint to obtain the information you desire.\
+**Note** that for security reasons some tables are private and some are public, you will only be provided with certain content.
+
+Your profile must have at least:
+
+- Basic user identification
+- XP amount
+- level
+- grades
+- skills
+- statistics
+
+### Instructions
+
+> Here is a small introduction and advantages of graphQL, if you want to read more about graphQL you can visit there [site](https://graphql.org/)
+
+GraphQL is a query language for any API endpoint and runtime. The syntax lets you specify what data you want to receive from that API.\
+You must have in mind that this language is used for API endpoints and not database. Unlike SQL,\
+graphql query does not go to your database directly but to your graphql API endpoint which can connect to a database or any other data source.
+
+You are already familiar with REST, since REST is a robust methodology of creating APIs and not elastic/scalable.\
+It can be at the same time painful, because it requires individual creation of each API, example: `v1/user/item/{id}`, `v1/post/item/{id}` and so on.
+
+The main feature of graphql compared to REST is that it lets you ask for specific information. And even better then that is the nesting feature.
+
+---
+
+### **profile**
+
+You will have to create a profile system where you can see all the info of a student. The information/data is present\
+ on the graphQL endpoint, where you will have to query it.
+
+The display of the information is up to you to design, but it must include:
+
+- **Basic user identification**, for example **githubLogin**
+- **XP** amount
+- **level**
+- **grades**
+- **skills**
+- **statistics**
+
+Any other information you desire to display is welcome and will be noted
+
+Beside this information, you will have to create a search engine which returns a selection of students profiles, based on the students:
```

Comment:
A search engine !?
How do we handle that on our side ?
I think it's more of a UI problem than a query problem, I don't know... I don't think graphQL is a good fit for a search type query system.

You can definitly do it but, still not sure of the overall value.